src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  max-width: (7 * $day-side) + (14 * $day-gap) + (2 * $calendar-padding);
+  padding: $calendar-padding;
+
+  @each $day, $index in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: ($day-side * $index) + ($day-gap * $index);
+    }
+  }
+
+  @for $lastDate from 28 through 31 {
+    &--month-length--#{$lastDate} {
+      .calendar__day:nth-child(n + #{$lastDate + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    @for $day from 1 through 31 {
+      &--#{$day}::before {
```

Comment:
here you can use `&:nth-child(#{$day})::before {`
and rid of enumerated classes (`calendar__day--31`)